File: SpigotLink/src/main/java/com/github/mrgeotech/SpigotLink.java
package com.github.mrgeotech;

import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.IOException;

public class SpigotLink extends JavaPlugin {

    private ChannelManager channelManager;

    @Override
    public void onEnable() {
        System.out.println("Enabling...");
        try {
            saveDefaultConfig();
            channelManager = new ChannelManager(this);
            Bukkit.getScheduler().runTaskTimer(this, channelManager, 1, 2);
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.out.println("Enabled!");
    }

    @Override
    public void onDisable() {
        System.out.println("Disabling...");
        channelManager.shutdown();
        Bukkit.getScheduler().cancelTasks(this);
        saveConfig();
        System.out.println("Disabled!");
    }

    public int getPort() {
        return getConfig().getInt("port");
    }

    public String getKey() {
        return getConfig().getString("key");
    }

}

File: SpigotLink/src/main/kotlin/net/mrgeotech/SpigotLink.kt
package net.mrgeotech

import net.mrgeotech.blockdata.BukkitBlockDataSupplier
import net.mrgeotech.network.ChunkServer
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin
import java.net.InetSocketAddress
import javax.net.ssl.SSLContext

class SpigotLink: JavaPlugin() {

    private lateinit var chunkServer: ChunkServer

    override fun onEnable() {
        saveDefaultConfig()

        chunkServer = ChunkServer(
            config.getInt("port", 10000),
            config.getStringList("trusted-addresses").map { InetSocketAddress(it, 10000).address.address },
            SSLContext.getDefault(),
            BukkitBlockDataSupplier(),
            this.logger
        ) {
            Bukkit.getScheduler().runTask(this, it)
        }

        Bukkit.getScheduler().runTaskAsynchronously(this, chunkServer::run)

        server.getPluginCommand("blockdata")?.setExecutor(BlockDataCommand())
    }

    override fun onDisable() {
        // Stop task
        Bukkit.getScheduler().cancelTasks(this)
        // Disconnect clients
        chunkServer.stop()
    }

}